addons/l10n_in_withholding/models/account_move.py

```diff
@@ -49,6 +50,46 @@ def _compute_l10n_in_total_withholding_amount(self):
             move.l10n_in_total_withholding_amount = sum(move.l10n_in_withhold_move_ids.filtered(
                 lambda m: m.state == 'posted').l10n_in_withholding_line_ids.mapped('l10n_in_withhold_tax_amount'))
 
+    def get_invalid_tax_records(self, check_moves=True):
+        """
+        :param check_move: If True, returns a list of invalid move IDs.
+                           If False, returns a list of invalid invoice line IDs.
+        """
+        invalid_tax_records = []
+        for move in self:
+            if move.country_code == 'IN' and not move.commercial_partner_id.l10n_in_pan:
```

Comment:
Hello @hamo-odoo ,
We can't add self.ensure_one() here because we used this method in `l10n_in_reports/models/l10n_in_report_handler.py` model also and there we got multiple records of `account.move`.

The same is true for the `if` condition. We used this method in both places, so it's better to add the condition here instead of in both models.

Reply:
Well I missed out to see the enterprise PR, I made some comments.
Then I would suggest in case `check_moves` is `True` you can simply return the move_id or just `True` in that case.

Reply:
and yes you can keep the conditions here but with `self.ensure_one()`